qemu/tests/nic_hotplug.py

```diff
@@ -122,40 +135,56 @@ def ping_hotplug_nic(ip, mac, session, is_linux_guest):
             nic_model = nic_params["pci_model"]
             nic_params["nic_model"] = nic_model
             nic_params["nic_name"] = nic_name
-
-            error.context("Disable other link(s) of guest", logging.info)
-            disable_nic_list = primary_nic + nic_hotplugged
-            for nic in disable_nic_list:
-                if guest_is_linux:
-                    ifname = utils_net.get_linux_ifname(s_session, nic["mac"])
-                    s_session.cmd_output_safe("ifconfig %s 0.0.0.0" % ifname)
+            used_sameid = params.get("used_sameid")
+
+            if used_sameid == "yes":
+                useddevice_id = primary_nic[0].netdev_id
+                logging.info("Hot-plug NIC with the netdev already in use")
+                try:
+                    add_output = device_add_nic(nic_model, useddevice_id, nic_name)
+                except Exception, err_msg:
+                    match_error = "it's in use"
+                    if match_error in str(err_msg):
+                        s_session.close()
+                        return
+                    else:
+                        logging.info("err_msg: %s" % str(err_msg))
+                        raise test.fail("Hot-plug error message is not as expected!")
                 else:
-                    s_session.cmd_output_safe("ipconfig /release all")
-                vm.set_link(nic.device_id, up=False)
-
-            error.context("Hotplug %sth '%s' nic named '%s'" % (nic_index,
+                    raise test.fail("Qemu should exit with error")
```

Comment:
@xiywang Sorry, I am confused by this error message, qemu should exit? Or you mean qemu should reject the operation with error?